server/datastore/cached_mysql/cached_mysql.go

```diff
@@ -313,10 +317,49 @@ func (ds *cachedMysql) DeleteTeam(ctx context.Context, teamID uint) error {
 	agentOptionsKey := fmt.Sprintf(teamAgentOptionsKey, teamID)
 	featuresKey := fmt.Sprintf(teamFeaturesKey, teamID)
 	mdmConfigKey := fmt.Sprintf(teamMDMConfigKey, teamID)
+	teamKey := fmt.Sprintf(teamByIdKey, teamID)
 
 	ds.c.Delete(agentOptionsKey)
 	ds.c.Delete(featuresKey)
 	ds.c.Delete(mdmConfigKey)
+	ds.c.Delete(teamKey)
 
 	return nil
 }
+
+func (ds *cachedMysql) Team(ctx context.Context, teamID uint) (*fleet.Team, error) {
```

Comment:
This is also used by the UI. So here too we'll need a separate `Team` method for the agents.

Reply:
https://github.com/fleetdm/fleet/pull/12815/commits/9d50bfb2c549f5d4840ed692bac11cb3cf71ce09